### DISCUSSION TITLE
Integrate templating engine

### DISCUSSION THREAD

**Discussion by mdicss** [Dynatrace/dynatrace-configuration-as-code#522] [Ideas]
Hi
It would be very helpful to have a templating engine (something like jinja in Ansible/python - maybe you know a better one) integrated into monaco. One possibility is to integrate this directly or to have a kind of preprocessor.
With this, it would be possible to have loops, conditions, generated variable values or to generate bigger parts of yaml or json configuration files.
My actual example are dashboards where I have a different number of charts (one for each host) in different environments.
See also: https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/discussions/410

My workaround is, to write a python skript to generate json files or to use python and jinja.

Regards, Matthias

**Comment by mdicss** [comment]
Any plans here?

**Comment by UnseenWizzard** [comment]
Hi @mdicss. 
No plans for this at the moment. 
Technically (but not really tested or supported by us!) monaco 2.0's complex value parameter [1] and control structures offered by the Go templating engine [2] could likely be used together to achieve what you want.

Disclaimer, I've only tried this for templating simple lists and due to the complexity of configuration we generally don't advice or support using more than the basic value replacement templating. 
But if you're already scripting around monaco this might be a usable solution for you :) 


[1] https://www.dynatrace.com/support/help/manage/configuration-as-code/configuration/yaml-configuration#value
[2] https://developer.hashicorp.com/nomad/tutorials/templates/go-template-syntax

**Reply by mdicss** [reply]
Thanks, but I think go templates are quite simple, too simple. To generate dashboards e.g. it is easier to use other template engines.
